### DISCUSSION TITLE
Is there a way to implement internal images?

### DISCUSSION THREAD

**Discussion by pllee1** [freeplane/freeplane#1411] [Help]
I like to share my mindmap with images to my students, but the images get lost, because they are just linked to the images on my computer. Is there a way to implement the images to the mindmap?
I found a freeplane handbook from 2018 and it says that there is a way to insert internal images but i dont find the way to do so in the newest version.. 
ty for your help.

**Comment by nnako** [comment]
Hi @pllee1 ,

images will currently not be included within the mindmap file but always be represented by a link which points to a file located map-external. There are ways to directly integrate any kind of files into mindmap files by using e.g. BASE64 encoding as seen in some email types. But this would have to be implemented.

If you are fine with a workaround, you could see the following post:
https://github.com/freeplane/freeplane/discussions/1373

Here, @macmarrum introduced an AddOn to "pack and go". There is also an external Python script mentioned to do such thing as packing all mindmap contents into one folder and zipping it. Thus, your students could unpack it and have every link working.

**Reply by beccon4** [reply]
Handling images in Freeplane is a mess -unfortunately. Links get broken when the map is moved or sent to someone else. It would be extremely helpful to have a way to include images. PackAndGo is a workaroud and probably the way to go - but it should be included into the main program and work seamlessly. Hope to get heard. :-)
